Tools/Cli-LoRa-Device-Provisioning/IoTDeviceHelper.cs

```diff
@@ -0,0 +1,1113 @@
+            if (string.Equals("null", opts.GatewayId, StringComparison.InvariantCultureIgnoreCase))
+            {
+                opts.GatewayId = string.Empty;
+                StatusConsole.WriteLine(MessageType.Info, $"GatewayId is set to \"null\". Adding empty property.");
+            }
+
+            if (string.Equals("null", opts.SensorDecoder, StringComparison.InvariantCultureIgnoreCase))
+            {
+                opts.SensorDecoder = string.Empty;
+                StatusConsole.WriteLine(MessageType.Info, $"SensorDecoder is set to \"null\". Adding empty property.");
+            }
+
+            return opts;
+        }
+
+        public bool ValidateDevice(Program.AddOptions opts, string fCntUpStartReported, string fCntDownStartReported, string fCntResetCounterReported, ConfigurationHelper configurationHelper)
+        {
+            string validationError = string.Empty;
+            bool isValid = true;
+
+            Console.WriteLine();
+            Console.WriteLine($"Verifying device {opts.DevEui} twin data...");
+
+            // ******************************************************************************************
+            // DevEui
+            // ******************************************************************************************
+            if (string.IsNullOrEmpty(opts.DevEui))
+            {
+                StatusConsole.WriteLine(MessageType.Error, "DevEui is missing.");
+                isValid = false;
+            }
+            else if (!ValidationHelper.ValidateHexStringTwinProperty(opts.DevEui, 8, out validationError))
+            {
+                StatusConsole.WriteLine(MessageType.Error, $"DevEui {opts.DevEui} is invalid: {validationError}.");
+                isValid = false;
+            }
+            else
+            {
+                StatusConsole.WriteLine(MessageType.Info, $"DevEui {opts.DevEui} is valid.");
+            }
+
+            // ******************************************************************************************
+            // ABP device specific properties
+            // ******************************************************************************************
+            if (string.Equals(opts.Type, "ABP", StringComparison.InvariantCultureIgnoreCase))
+            {
+                // NwkSKey
+                if (string.IsNullOrEmpty(opts.NwkSKey))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, "NwkSKey is missing.");
+                    isValid = false;
+                }
+                else if (!ValidationHelper.ValidateHexStringTwinProperty(opts.NwkSKey, 16, out validationError))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, $"NwkSKey {opts.NwkSKey} is invalid: {validationError}.");
+                    isValid = false;
+                }
+                else
+                {
+                    StatusConsole.WriteLine(MessageType.Info, $"NwkSKey {opts.NwkSKey} is valid.");
```

Comment:
That's the only "ok" message, the others don't print that. I think we should remove this as well?

Reply:
I don't understand this one. Let's discuss.

Reply:
Discussed.